# HG changeset patch
# User Olaf Till <olaf.till@uni-jena.de>
# Date 1329313477 -3600
#      Wed Feb 15 14:44:37 2012 +0100
# Branch stable
# Node ID 07c55bceca23e36e2d37dbfe0e6e5df9f9ddaaa3
# Parent  cb4f1915db9210f297dc8d16516ee4366edd2259
Fix guarded_eval() subfunction in fminunc (bug #35534).

* fminunc.m: Fix guarded_eval() subfunction in fminunc (bug #35534).

diff --git a/scripts/optimization/fminunc.m b/scripts/optimization/fminunc.m
--- a/scripts/optimization/fminunc.m
+++ b/scripts/optimization/fminunc.m
@@ -345,20 +345,18 @@ endfunction
 function [fx, gx] = guarded_eval (fun, x)
   if (nargout > 1)
     [fx, gx] = fun (x);
   else
     fx = fun (x);
     gx = [];
   endif
 
-  if (! (isreal (fx) && isreal (jx)))
+  if (! (isreal (fx) && isreal (gx)))
     error ("fminunc:notreal", "fminunc: non-real value encountered");
-  elseif (complexeqn && ! (isnumeric (fx) && isnumeric(jx)))
-    error ("fminunc:notnum", "fminunc: non-numeric value encountered");
   elseif (any (isnan (fx(:))))
     error ("fminunc:isnan", "fminunc: NaN value encountered");
   endif
 endfunction
 
 %!function f = __rosenb (x)
 %!  n = length (x);
 %!  f = sumsq (1 - x(1:n-1)) + 100 * sumsq (x(2:n) - x(1:n-1).^2);
